jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/error/JobAutoAckErrorCommand.java

```diff
@@ -40,22 +43,29 @@
  */
 public class JobAutoAckErrorCommand extends AutoAckErrorCommand {
 
+    private static final Logger logger = LoggerFactory.getLogger(JobAutoAckErrorCommand.class);
+
     private static final String RULE = "Jobs that previously failed but now are in one of the statuses - queued, completed or cancelled";
     
     @SuppressWarnings("unchecked")
     @Override
     protected List<ExecutionErrorInfo> findErrorsToAck(EntityManager em) {
-        String findJobErrorsQuery = "select error from ExecutionErrorInfo error "
-                + "where error.type = :type "
-                + "and error.acknowledged =:acknowledged "
-                + "and error.jobId in (select req.id from RequestInfo req where status in (:status))";
-        
-        List<ExecutionErrorInfo> errorsToAck = em.createQuery(findJobErrorsQuery)
-                .setParameter("type", JobExecutionErrorFilter.TYPE)
-                .setParameter("acknowledged", new Short("0"))
-                .setParameter("status", Arrays.asList(STATUS.DONE, STATUS.CANCELLED, STATUS.QUEUED))
-                .getResultList();
+        List<ExecutionErrorInfo> errorsToAck = new ArrayList<>();
+        em.getTransaction().begin();
```

Comment:
this implies you use a resource local entity manager and thus might cause issues when there is actually JTA entity manager being used. Have this been tested on EAP/WAS/WLS?

Reply:
just using the transaction manager coming from drools